ui/src/app/views/workspace-edit/component.tsx

```diff
@@ -449,78 +532,189 @@ export const WorkspaceEdit = fp.flow(withRouteConfigData(), withCurrentWorkspace
-              <WorkspaceCategory item={ResearchPurposeItems.aggregateAnalysis}
-                  value={!!this.state.workspace.researchPurpose.aggregateAnalysis}
-                  onChange={v => this.updateWorkspaceCategory('aggregateAnalysis', v)}/>
+                 value={this.state.workspace.researchPurpose.methodsDevelopment}
+                 onChange={v => this.updateResearchPurpose('methodsDevelopment', v)}/>
             </div>
-          </div>
-          <div style={{display: 'inline-block'}}>
-            <div style={{display: 'flex'}}>
+            <div style={{display: 'flex', flexDirection: 'column'}}>
               <WorkspaceCategory item={ResearchPurposeItems.controlSet}
-                  value={this.state.workspace.researchPurpose.controlSet}
-                  onChange={v => this.updateWorkspaceCategory('controlSet', v)}/>
-              <WorkspaceCategory item={ResearchPurposeItems.ancestry}
-                  value={this.state.workspace.researchPurpose.ancestry}
-                  onChange={v => this.updateWorkspaceCategory('ancestry', v)}/>
+                 value={this.state.workspace.researchPurpose.controlSet}
+                 onChange={v => this.updateResearchPurpose('controlSet', v)}/>
+              <WorkspaceCategory item={ResearchPurposeItems.educational}
+                 value={this.state.workspace.researchPurpose.educational}
+                 onChange={v => this.updateResearchPurpose('educational', v)}/>
+              <WorkspaceCategory item={ResearchPurposeItems.commercialPurpose}
+                 value={this.state.workspace.researchPurpose.commercialPurpose}
+                 onChange={v => this.updateResearchPurpose('commercialPurpose', v)}/>
+              <WorkspaceCategory item={ResearchPurposeItems.otherPurpose}
+                 value={this.state.workspace.researchPurpose.otherPurpose}
+                 onChange={v => this.updateResearchPurpose('otherPurpose', v)}>
+                <TextArea value={this.state.workspace.researchPurpose.otherPurposeDetails}
+                          onChange={v => this.updateResearchPurpose('otherPurposeDetails', v)}
+                          disabled={!this.state.workspace.researchPurpose.otherPurpose}
+                          style={{marginTop: '0.5rem'}}/>
+              </WorkspaceCategory>
             </div>
           </div>
-          <div style={{display: 'inline-block'}}>
-            <div style={{display: 'flex'}}>
-              <WorkspaceCategory item={ResearchPurposeItems.population}
-                  value={this.state.workspace.researchPurpose.population}
-                  onChange={v => this.updateWorkspaceCategory('population', v)}/>
-              <WorkspaceCategory item={ResearchPurposeItems.commercialPurpose}
-                  value={this.state.workspace.researchPurpose.commercialPurpose}
-                  onChange={v => this.updateWorkspaceCategory('commercialPurpose', v)}/>
+        </WorkspaceEditSection>
+        <WorkspaceEditSection
```

Comment:
This especially is where I feel like there is value in the map. All of these seem like repeated code.